pkg/uhttp/wrapper.go

```diff
@@ -128,19 +147,32 @@ func WithResponse(response interface{}) DoOption {
 }
 
 func (c *BaseHttpClient) Do(req *http.Request, options ...DoOption) (*http.Response, error) {
+	var cacheKey string = GetCacheKey(req)
```

Comment:
We should only generate the cache key if the request method is GET.

Reply:
Fixed